src/components/createAuthorModal/CreateAuthor.tsx

```diff
@@ -0,0 +1,63 @@
+import { Dispatch, SetStateAction, useState } from 'react';
+
+import AuthorRequest from '../../interfaces/AuthorRequest';
+import { createAuthor } from '../../services/author.service';
+import './CreateAuthor.css';
+
+interface Props {
+  show : boolean;
+  closeModal: () => void;
+  setIsAuthorsChanged : Dispatch<SetStateAction<boolean>>
+}
+
+function CreateAuthorModal(props: Props) {
+  const [ author, setAuthor ] = useState<AuthorRequest>({ FirstName: '', LastName: '' })
+  const [ errorMessage, setErrorMessage ] = useState('')
+
+  const create = () => {
+    if(validate()) {
+      createAuthor(author).then(() => {
+        props.closeModal()
+        props.setIsAuthorsChanged(true)
+      })
+        .catch(() => {setErrorMessage('Unable to create new author.')})
+    }
+  }
+  const validate = () : boolean => {
+    if(author.FirstName.trim() === '' || author.LastName.trim() === '') {
+      setErrorMessage('Please fill all the fields.')
+      return false
+    }
+    return true
+  }
+
+  if (!props.show) return null
```

Comment:
You can manage show / hide modal outside of the modal component instead of returning null here

Reply:
Also, your file is called CreateAuthor and it's function is called CreateAuthorModal, you should use the same name for both to avoid confusion